### ISSUE TITLE
AI: Crafter escapes task loop and then enters another 

### ISSUE THREAD

**Issue by com1clyf3** [CompletelyFairGames/dwarfcorp#754]
![image](https://user-images.githubusercontent.com/6562812/40683310-390cbcf4-635c-11e8-98ef-4620c2957150.png)

**TLDR: A crafter is stuck in one loop. Escapes, and then enters another. Ad infinitim. Can't make a save of this for some reason.**

The crafter (Lehheg Pickpriest) was found at the bottom of the dig pit cycling in a loop between stock tasks - emerald, ice, and rock. It continued this loop for about a minute on 3x, and then escaped.  It behaved normally for about ten seconds and then entered another loop.

This loop had to do with trying to complete some build tasks, but those kept failing. He stood in place.

Then it was time to sleep. He cycled inbetween sleep, getting up to do build a barrel, and then going back to sleep.

Eventually he bailed on sleep and stood in place contemplating which barrel to build.
![image](https://user-images.githubusercontent.com/6562812/40683853-b9c68824-635d-11e8-9956-7ea03b04a24a.png)

Tried to get a save of, but saves are failing on this for some reason:
![image](https://user-images.githubusercontent.com/6562812/40684329-03cbc0b4-635f-11e8-8a77-1452f1dcc9c5.png)

Failed save:
[Insect Realm_60_131720999736494581.zip](https://github.com/CompletelyFairGames/dwarfcorp/files/2050806/Insect.Realm_60_131720999736494581.zip)
World:
[Insect Realm.zip](https://github.com/CompletelyFairGames/dwarfcorp/files/2050808/Insect.Realm.zip)

5c6a30a6

**Comment by mklingen** [comment]
The only way I can debug this is with a save. The fact that saving is failing is the real bug here. Can you please locate the `log.txt` entry relating to the failed save?

**Comment by com1clyf3** [comment]
>    at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
>    at DwarfCorp.FileUtils.Save[T](JsonSerializer serializer, T obj, String filePath, Boolean compress) in C:\Users\kingofarabi1\Source\Repos\dwarfcorp\DwarfCorp\DwarfCorpXNA\AssetManagement\FileUtils-SaveJson.cs:line 169
>    at DwarfCorp.SaveGame.WriteFile(String directory) in C:\Users\kingofarabi1\Source\Repos\dwarfcorp\DwarfCorp\DwarfCorpXNA\AssetManagement\GameSave\SaveGame.cs:line 55
>    at DwarfCorp.WorldManager.SaveThreadRoutine(String filename) in C:\Users\kingofarabi1\Source\Repos\dwarfcorp\DwarfCorp\DwarfCorpXNA\World\WorldManager.cs:line 671Overworld C:\Users\kingofarabi1\AppData\Roaming\DwarfCorp\Worlds\Insect Realm\world.png already exists. Just assuming it is correct.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at DwarfCorp.Task.Equals(Object obj) in C:\Users\kingofarabi1\Source\Repos\dwarfcorp\DwarfCorp\DwarfCorpXNA\Scripting\TaskManagement\Task.cs:line 118
>    at System.Collections.Generic.ObjectEqualityComparer`1.Equals(T x, T y)
>    at System.Collections.Generic.List`1.Contains(T item)

**Comment by mklingen** [comment]
okay, this is a duplicate of #729

**Comment by com1clyf3** [comment]
Ah. Will close this then.

**Comment by mklingen** [comment]
The save issue is a duplicate, not whatever this AI issue is.

**Comment by com1clyf3** [comment]
https://steamcommunity.com/app/252390/discussions/0/1728701877499084571/

Probably #762 , but figured I'd link it here.
